### ISSUE TITLE
Get test loss

### ISSUE THREAD

**Issue by nhaxin204** [AlexeyAB/darknet#4612]
I want to get the loss of the test sample. What should I do！

**Comment by barzan-hayati** [comment]
> I want to get the loss of the test sample. What should I do！

For validation data you could use **map** function.

**Comment by AlexeyAB** [comment]
@nhaxin204 I did not find a single reason why a test loss may be better than accuracy (mAP).

So to get train-val and test-map charts you can train with ` -map` flag at the end of training command: https://github.com/AlexeyAB/darknet#when-should-i-stop-training
![68747470733a2f2f6873746f2e6f72672f776562742f79642f766c2f61672f7964766c616775746f66327a636e6a6f64737467726f656e3861632e6a706567](https://user-images.githubusercontent.com/4096485/71694590-ac8b4300-2dc0-11ea-8d9b-cfc3fcb79590.jpg)

**Comment by barzan-hayati** [comment]
> 
> 
> @nhaxin204 I did not find a single reason why a test loss may be better than accuracy (mAP).
> 
> So to get train-val and test-map charts you can train with ` -map` flag at the end of training command: https://github.com/AlexeyAB/darknet#when-should-i-stop-training
> ![68747470733a2f2f6873746f2e6f72672f776562742f79642f766c2f61672f7964766c616775746f66327a636e6a6f64737467726f656e3861632e6a706567](https://user-images.githubusercontent.com/4096485/71694590-ac8b4300-2dc0-11ea-8d9b-cfc3fcb79590.jpg)

I think @nhaxin204  wants to check **loss** and **map** for **test data** _not_ for **validation data**. By `-map` we could check **loss** for **training** and by `map `we could check **mAP** for **validation** data.

**Comment by AlexeyAB** [comment]
@barzan-hayati 
>  map for test data not for validation data

What is the difference between test data and validation data?

**Comment by nhaxin204** [comment]
@AlexeyAB I want to delete bad samples through the unique loss

**Comment by AlexeyAB** [comment]
@nhaxin204 Do you want to get test-loss for each image?

**Comment by nhaxin204** [comment]
@AlexeyAB Yes, I want to get the loss of each image!

**Comment by lvpchen** [comment]
I found map is increasing constantly and train loss is stably, when i training my custom dataset. How can i judge overfitting from map-loss curve.
![Uploading chart.png…]()
